server/views.py

```diff
@@ -182,6 +182,48 @@ def card(request, tool_id, card_id):
 
   return makeResponse(request, result)
 
+@check_secret
+@check_ip
+@require_GET
+def memberid(request, tool_id, card_id):
+  ip = get_ip(request)
+  try:
+    t = Tool.objects.get(pk=tool_id)
+  except ObjectDoesNotExist as e:
+    logger.warning('Tool does not exist for %s // %s from %s', request.method, request.path, ip,
+                extra={
+                    'status_code': 200,
+                    'request': request
+                }
+            )
+    result = { 'numeric_status' : -1, 'error' : 'Tool does Not Exist' }
+    return makeResponse(request, result)
+  try:
+    c = Card.objects.get(card_id=card_id)
+  except ObjectDoesNotExist as e:
+    logger.warning('Card does not exist for %s // %s from %s', request.method, request.path, ip,
+                extra={
+                    'status_code': 200,
+                    'request': request
+                }
+            )
+    result = { 'numeric_status' : -1, 'error' : 'Card does Not Exist' }
+    return makeResponse(request, result)
+
+  result = { 'numeric_status' : -1 }
+  perm_text = 'Unknown'
+
+  if not c.user.subscribed:
+    result['numeric_status'] = -1
+    perm_text = 'Not Subscribed'
+    return makeResponse(request, result)
+  else:
+    # For subscribed users, getting the name is useful for doorbot
+    result['numeric_status'] = c.user.id
```

Comment:
numeric_status is a somewhat broken idea. Use the JSON mode for less pain

Reply:
Removed the memberid function and instead added a user_id field to the json response for the card function.